### DISCUSSION TITLE
Feature: Jump directory using z for fish

### DISCUSSION THREAD

**Discussion by roele** [PatrickF1/fzf.fish#231] [Ideas]
As some might find it handy i'll share my custom function to integrate with https://github.com/jethrokuan/z.

```shell
#
# fzf jump directory function
#
# requires: https://github.com/jethrokuan/z
# __________________________
#

function _fzf_jump_directory

  if not type -q z
    set_color red
      echo "_fzf_jump_directory: z not found." >&2
    set_color normal
    return 1
  end

  set current_token (commandline --current-token)

  set command_z (
    z -l | sort -rn | cut -c 12- | _fzf_wrapper --query=$current_token $fzf_jump_directory_opts
  )

  if test $status -eq 0
    cd $command_z
  end

  commandline --function repaint

end
```

Once defined you can bind the function to keys of your preference.

```shell
function fish_user_key_bindings
  bind \cz _fzf_jump_directory            # Ctrl-z
end
```

**Comment by dgmulf** [comment]
Brilliant! Thank you so much for this contribution. It's exactly what I was searching for. A nice additional feature would be to implement the same directory contents preview used in the built-in picker.

**Reply by roele** [reply]
@dgmulf Not sure what built-in picker you're referring to but adding a directory preview is pretty stright forward. Just append `--preview='ls -alh {1}'` to the `_fzf_wrapper` command. If you're using [exa](https://github.com/gazorby/fish-exa) you could also show a tree view (`lt 2 {1}`) for example.

**Reply by dgmulf** [reply]
@roele Thanks! I was referring to `_fzf_search_directory.fish`, included in the core plugin. I just copied the preview argument as it appears there, `--preview='_fzf_preview_file {}'`, for the sake of consistency, and it appears to work properly.

**Comment by PatrickF1** [comment]
If you use https://github.com/ajeetdsouza/zoxide, you can get something very similar when you run `zi`.
